From: Sune Vuorela <sune@vuorela.dk>
Date: Sat, 29 Dec 2012 23:56:27 +0100
Subject: create a system default path

Make it possible to build qtchooser with a system default search path

build using make QTCHOOSER_GLOBAL_DIR=/usr/share/ will make qtchooser
fall back to search in /usr/share/qtchooser/ as a last option

Modified by Lisandro Damián Nicanor Pérez Meyer.
This patch has not been accepted upstream (see link below), but
we are able to ship it until we create a better patch for our use case.

Origin: upstream, https://codereview.qt-project.org/#change,44027
Last-Update: 2013-03-20

--- a/src/qtchooser/Makefile
+++ b/src/qtchooser/Makefile
@@ -17,6 +17,10 @@ TARGET        = qtchooser
 OBJECTS_TEST  = main-test.o
 TARGET_TEST   = test/qtchooser
 
+ifneq ($(QTCHOOSER_GLOBAL_DIR),)
+	QTCHOOSER_GLOBAL_DIR_VAR:=-DQTCHOOSER_GLOBAL_DIR=\"$(QTCHOOSER_GLOBAL_DIR)\"
+endif
+
 first: all
 check: $(TARGET_TEST)
 
@@ -49,10 +53,10 @@ uninstall:
 ####### Compile
 
 main.o: main.cpp
-	$(CXX) -c -Wall -Wextra $(CXXFLAGS) $(INCPATH) -o main.o main.cpp
+	$(CXX) -c -Wall -Wextra $(QTCHOOSER_GLOBAL_DIR_VAR) $(CXXFLAGS) $(INCPATH) -o main.o main.cpp
 
 main-test.o: main.cpp
-	$(CXX) -c -Wall -Wextra -DQTCHOOSER_TEST_MODE -g $(CXXFLAGS) $(INCPATH) -o main-test.o main.cpp
+	$(CXX) -c -Wall -Wextra -DQTCHOOSER_TEST_MODE $(QTCHOOSER_GLOBAL_DIR_VAR) -g $(CXXFLAGS) $(INCPATH) -o main-test.o main.cpp
 
 ####### Install
 
--- a/src/qtchooser/main.cpp
+++ b/src/qtchooser/main.cpp
@@ -265,6 +265,12 @@ vector<string> ToolWrapper::searchPaths(
     // search the XDG config location directories
     paths = stringSplit(qgetenv("XDG_CONFIG_DIRS", "/etc/xdg").c_str());
 
+#if defined(QTCHOOSER_GLOBAL_DIR)
+    // Add QTCHOOSER_GLOBAL_DIR at the beginning.
+    vector<string> globalDir = stringSplit(QTCHOOSER_GLOBAL_DIR);
+    paths.insert(paths.begin(), globalDir.begin(), globalDir.end());
+#endif
+
     string localDir = qgetenv("XDG_CONFIG_HOME", userHome() + PATH_SEP ".config");
     paths.push_back(localDir);
 
